fsharp-backend/src/BackendOnlyStdLib/HttpQueryEncoding.fs

```diff
@@ -0,0 +1,125 @@
+module BackendOnlyStdLib.HttpQueryEncoding
+
+
+type AspHeaders = System.Net.Http.Headers.HttpHeaders
+
+open Prelude
+open LibExecution
+open LibBackend
+open VendoredTablecloth
+
+
+module RT = RuntimeTypes
+
+// -------------------------
+// URLs and queryStrings
+// -------------------------
+
+// For putting into URLs as query params
+let rec toUrlString (dv : RT.Dval) : string =
+  let r = toUrlString
+  match dv with
+  | RT.DFnVal _ ->
+    (* See docs/dblock-serialization.ml *)
+    "<block>"
+  | RT.DIncomplete _ -> "<incomplete>"
+  | RT.DPassword _ -> "<password>"
+  | RT.DInt i -> string i
+  | RT.DBool true -> "true"
+  | RT.DBool false -> "false"
+  | RT.DStr s -> s
+  | RT.DFloat f -> LibExecution.DvalReprExternal.ocamlStringOfFloat f
+  | RT.DChar c -> c
+  | RT.DNull -> "null"
+  | RT.DDate d -> RT.DDateTime.toIsoString d
+  | RT.DDB dbname -> dbname
+  | RT.DErrorRail d -> r d
+  | RT.DError _ -> "error="
+  | RT.DUuid uuid -> string uuid
+  | RT.DHttpResponse (RT.Redirect _) -> "null"
+  | RT.DHttpResponse (RT.Response (_, _, hdv)) -> r hdv
```

Comment:
can we rename this? I don't know what hdv is supposed to be

Reply:
I'd rather not touch that code. `hdv` is the http response dval - the dval contained in the http response (as opposed to the dval that contains it).